pkg/server/resource_lister.go

```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type ResourceLister struct {
+	ResourcesPath string
+	BearerToken   string
+	K8sEndpoint   string
+	Client        *http.Client
+}
+
+func (l *ResourceLister) handleResources(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		sendResponse(w, http.StatusMethodNotAllowed, apiError{"Invalid method: only GET is allowed"})
+		return
+	}
+	req, err := http.NewRequest("GET", l.K8sEndpoint+l.ResourcesPath, nil)
+	if err != nil {
+		sendResponse(w, http.StatusInternalServerError, apiError{fmt.Sprintf("Failed to create configmap for monitoring dashboard GET request: %v", err)})
```

Comment:
I would make these responses more generic in case we use `ResourceLister` for other things. There's nothing in this file that's specific to the dashboard config maps.